### PR TITLE
[vrotsc] (#318) Item presentation is now better structured (tree)

### PR DESCRIPTION
**Pull request by Michaelpalacce**
### Description

image

### Checklist

- [x] I have added relevant error handling and logging messages to help troubleshooting
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation, relevant usage information (if applicable)
- [x] I have updated the PR title with affected component, related issue number and a short summary of the changes introduced
- [x] I have added labels for implementation kind (kind/*) and version type (version/*)
- [x] I have tested against live environment, if applicable
- [x] I have synced any structure and/or content vRA-NG improvements with vra-ng and ts-vra-ng archetypes (if applicable)
- [x] I have my changes rebased and squashed to the minimal number of relevant commits. **Notice: don't squash all commits**
- [x] I have added a descriptive commit message with a short title, including a `Fixed #XXX -` or `Closed #XXX -` prefix to auto-close the issue

### Testing

e2e test added

### CODE REVIEW

File: typescript/vrotsc/e2e/cases/canvas-items/waiting-timer-edge-2.wf.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { Workflow, In, WaitingTimerItem } from "vrotsc-annotations";
 
 @Workflow({
-	name: "Waitin Timer Edge 2",
+	name: "Waitin Timer Edge",
```
Comment (bcpmihail) [inline typescript/vrotsc/e2e/cases/canvas-items/waiting-timer-edge-2.wf.ts:4]: * Waiting?

File: typescript/vrotsc/e2e/expect/canvas-items/workflows/VMware/PSCoE/Waitin Timer Edge.element_info.xml [hunk 2]
```diff
@@ -3,7 +3,7 @@
 <properties>
 <comment>Generated by vrotsc</comment>
 <entry key="categoryPath">VMware.PSCoE</entry>
-<entry key="name">Waiting Timer Edge</entry>
+<entry key="name">Waitin Timer Edge</entry>
```
Comment (bcpmihail) [inline typescript/vrotsc/e2e/expect/canvas-items/workflows/VMware/PSCoE/Waitin Timer Edge.element_info.xml:6]: \* Waiting?

File: typescript/vrotsc/src/compiler/transformer/fileTransformers/workflow/decorators/helpers/graph.ts [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+ * Copyright 2023 VMware, Inc.
+ *
+ * This product is licensed to you under the BSD-2 license (the "License"). You may not use this product except in compliance with the BSD-2 License.
+ *
+ * This product may include a number of subcomponents with separate copyright notices and license terms. Your use of these subcomponents is subject to the terms and conditions of the subcomponent's license, as noted in the LICENSE file.
+ * #L%
+ */
+
+/**
+ * The node holds information about it's position
```
Comment (bcpmihail) [inline typescript/vrotsc/src/compiler/transformer/fileTransformers/workflow/decorators/helpers/graph.ts:17]: \* its

File: typescript/vrotsc/src/compiler/transformer/fileTransformers/workflow/decorators/helpers/graph.ts [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+ */
+export interface GraphNode {
+	name: string;
+	x?: number;
+	y?: number;
+	targets: string[];
+}
+
+/**
+ * Leaf holds the node and it's children(leaves)
```
Comment (bcpmihail) [inline typescript/vrotsc/src/compiler/transformer/fileTransformers/workflow/decorators/helpers/graph.ts:27]: \* its

File: typescript/vrotsc/src/compiler/transformer/fileTransformers/workflow/decorators/helpers/graph.ts [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+	node: GraphNode;
+	leaves: Leaf[];
+}
+
+/**
+ * Tree holds the structure of the graph, it's populated with the calculated positions
+ */
+interface Tree {
+	startNodes: string[];
+	leaves: {
```
Comment (bcpmihail) [inline typescript/vrotsc/src/compiler/transformer/fileTransformers/workflow/decorators/helpers/graph.ts:39]: Can be Record<string, Leaf>

File: typescript/vrotsc/src/compiler/transformer/fileTransformers/workflow/decorators/helpers/graph.ts [hunk 6]
```diff
@@ -0,0 +1,334 @@
...
+	{ name: "T", targets: ["U", "W", "D"] },
+	{ name: "U", targets: [] },
+	{ name: "W", targets: [] },
+
+	// Third Start
+	{ name: "X", targets: ["Y"] },
+	{ name: "Y", targets: [] },
+];
+
+const graph = new Graph(nodes, ["A", "S", "X"]);
```
Comment (bcpmihail) [inline typescript/vrotsc/src/compiler/transformer/fileTransformers/workflow/decorators/helpers/graph.ts:79]: I don't see the 'X' start in the output 'diagram' below

File: typescript/vrotsc/src/compiler/transformer/fileTransformers/workflow/decorators/helpers/graph.ts [hunk 7]
```diff
@@ -0,0 +1,334 @@
...
+ * ```
+....................................................................................................
+....................................................................................................
+....................................................................................................
+..............................F....O....P....Q......................................................
+....................................................................................................
+....................................................................................................
+....................D....E..........................................................................
+....................................................................................................
+.....A....B....C...................J................................................................
```
Comment (bcpmihail) [inline typescript/vrotsc/src/compiler/transformer/fileTransformers/workflow/decorators/helpers/graph.ts:94]: The placement of 'J' with root 'I' is confusing, I think it needs to be in the row below

Comment (Michaelpalacce) [reply typescript/vrotsc/src/compiler/transformer/fileTransformers/workflow/decorators/helpers/graph.ts:94]: The calculations are not perfect here since they go down to fraction... rounding them up leads to deviations... look at the example for  C and D,G it's reversed... vRO somewhat supports this tho, so we are all good ... even tho vRO has it's own problems and it converts the calculations for x and y for some reason..... so it's not perfect, but it will do

File: docs/versions/latest/Release.md [hunk 8]
```diff
@@ -24,6 +24,72 @@
...
+  { name: "U", targets: [] },
+  { name: "W", targets: [] },
+  { name: "X", targets: ["Y"] },
+  { name: "Y", targets: [] },
+];
+```
+
+We get:
+```log
+....................................................................................................
```
Comment (bcpmihail) [inline docs/versions/latest/Release.md:66]: I know the idea here is to show the node positions, but on the diagram isn't very clear where the start nodes are and which branch a leaf belongs to (e.g. 'J' comes from I but it's on the same line as C)

File: typescript/vrotsc/src/compiler/transformer/fileTransformers/workflow/decorators/helpers/graph.ts [hunk 9]
```diff
@@ -0,0 +1,334 @@
...
+		this.calculated = true;
+
+		for (const leaf of Object.values(this.tree.leaves)) {
+			if (this.tree.startNodes.includes(leaf.node.name)) {
+				continue;
+			}
+
+			if (leaf.node.x === undefined || leaf.node.y === undefined) {
+
+				console.log(
```
Comment (bcpmihail) [inline typescript/vrotsc/src/compiler/transformer/fileTransformers/workflow/decorators/helpers/graph.ts:318]: Extract to a variable, e.g:
```
const nodes = Object.values(this.tree.leaves).map(l => ({ node: l.node, leaves: l.leaves.map(le => le.node.name) }));
console.log(JSON.stringify(nodes, null, 2));
```
Wondering whether it wouldn't be better to include the nodes in the error instead:
```
throw new Error(`Node "${leaf.node.name}" has no position and is possibly isolated. Check the WF element targets:\n`
+ JSON.stringify(nodes, null, 2));
```

File: typescript/vrotsc/src/compiler/transformer/fileTransformers/workflow/decorators/helpers/graph.ts [hunk 10]
```diff
@@ -0,0 +1,334 @@
...
+				return true;
+			}
+		}
+		return false;
+	}
+
+	/**
+	 * Builds the tree
+	 *
+	 * Calculates the x and y positions of each leaf
```
Comment (bcpmihail) [inline typescript/vrotsc/src/compiler/transformer/fileTransformers/workflow/decorators/helpers/graph.ts:237]: Add @param descriptions

**Comment by Michaelpalacce** [comment]
This also fixes #364

File: typescript/vrotsc/src/compiler/transformer/fileTransformers/workflow/decorators/helpers/graph.ts (hunk 9 above)
Comment (Michaelpalacce) [reply typescript/vrotsc/src/compiler/transformer/fileTransformers/workflow/decorators/helpers/graph.ts:330]: Error gets drowned by the nodes, that's why it's separate

File: package-installer/src/main/java/com/vmware/pscoe/iac/installer/Installer.java [hunk 11]
```diff
@@ -252,6 +252,18 @@ enum Option {
...
+	 * VRANG unrelease blueprint versions. Decides wether old versions need to be
+	 * unrelased
+	 *
+	 * This only works when running in non interactive mode
+	 */
+	VRANG_BP_UNRELEASE_VERSIONS(
+			"vrang_bp_unrelease_versions",
+			ConfigurationVraNg.UNRELEASE_BLUEPRINT_VERSIONS),
+
+
```
Comment (akantchev) [inline package-installer/src/main/java/com/vmware/pscoe/iac/installer/Installer.java:266]: Are you sure that this is a part of this PR, I think you need to revert this, and make it a part of another PR in order not to pollute the logic of this one.

Comment (Michaelpalacce) [reply package-installer/src/main/java/com/vmware/pscoe/iac/installer/Installer.java:266]: It's really really small... and I didn't want to go and create a special Pr for it. Can we make an exception?

**Review by bcpmihail — APPROVED** [review]
